doc/api.md

```diff
@@ -742,6 +744,60 @@ Unreserves a path for a publishing application. Returns success or failure only.
 - If a path reservation exists for the supplied base_path but a different
   publishing_app, the command will fail.
 
+## `PUT /publish-intent(/*base_path)`
+
+[Request/Response detail][put-intent-pact]
+
+Used to create or update a publishing intent for a particular base_path and
+reserve the base path in the publishing-api. A publishing intent sets when the TTL
+cache should be cleared.
+
+### Path parameters
+
+- `base_path`
+  - Identifies the path to create the publishing intent for
+
+### JSON parameters:
+
+- `publishing_app` *(required)*
+  - The name of the application making this request, words separated with
+    hyphens.
+- `rendering_app` *(required)*
+  - The name of the application rendering the content the base path refers to,
+    words separated with hyphens.
```

Comment:
This looks helpful, but I'm curious where the hyphen rule is coming from?

Reply:
I'm not sure exactly where it originates from, probably in the router so it can find which app to send the requests to when they come in.

`rendering_app` is one of the fields used to populate the PublishingIntent record in content-store which then populates the `backend_id` field if a new route is created (depending on presence of routes argument) in the router-api.

Reply:
Sorry if this is really small, you may have to get your specs on! 👁 

This shows the pattern of using hyphen separated words for the frontend app names...

![Screen Shot 2019-03-20 at 16 13 48](https://user-images.githubusercontent.com/24479188/54700563-2badd400-4b2b-11e9-8b7a-070733860bff.png)

Reply:
As per our discussion the rendering_app is validated against a regex which only allows lowercase, numbers and hyphens. So the names don't necessarily need to be seperated by hyphens so i will remove the rule! :)